libraries/common/array.effekt

```diff
@@ -62,8 +98,19 @@ extern global def unsafeGet[T](arr: Array[T], index: Int): T =
   js "${arr}[${index}]"
   chez "(vector-ref ${arr} ${index})"
   llvm """
-    %z = call %Pos @c_array_get(%Pos ${arr}, %Int ${index})
-    ret %Pos %z
+    %pos_tag = extractvalue %Pos ${arr}, 0
+    %pos_data = extractvalue %Pos ${arr}, 1
+
+    %data_pointer = getelementptr inbounds ptr, ptr %pos_data, i64 3
+    %element_pointer = getelementptr inbounds %Pos, ptr %data_pointer, i64 ${index}
+    %element_tag = load i64, ptr %element_pointer, align 8
+    %element_data_pointer = getelementptr inbounds ptr, ptr %element_pointer, i64 1
+    %element_data = load ptr, ptr %element_data_pointer, align 8
+    tail call void @sharePositive(i64 %element_tag, ptr %element_data)
+    tail call void @erasePositive(i64 %pos_tag, ptr %pos_data)
```

Comment:
Related to a previous comment on the parameters of `@sharePositive` and `@erasePositive`, shouldn't it be possible to load the entire `%Pos` in one load instruction? (Which will probably be lowered to two actual load instructions, but still....

Reply:
I will try

Reply:
Included in the newest version. This was possible in a lot of places and makes the code way cleaner